tests/unit/test_routes/test_resource_crud.py

```diff
@@ -0,0 +1,552 @@
+
+    # Bogus Data
+    name = False
+    url = "htt://bad_url.doesnotexist"
+    category = True
+    languages = False
+    paid = "Bad Data"
+    notes = True
+    response = update_resource(client,
+                               apikey,
+                               name,
+                               url,
+                               category,
+                               languages,
+                               paid,
+                               notes)
+    assert (response.status_code == 422)
+
+    # A String to Big for the DB
+    long_string = "x" * 6501
+    name = long_string
+    url = long_string
+    category = long_string
+    languages = long_string
+    paid = True
+    notes = long_string
+    response = update_resource(client,
+                               apikey,
+                               name,
+                               url,
+                               category,
+                               languages,
+                               paid,
+                               notes)
+    assert (response.status_code == 422)
+
+    # Unicode Characters
+    name = "😀"
+    url = None
+    category = None
+    languages = None
+    paid = True
+    notes = "∞"
+    response = update_resource(client,
+                               apikey,
+                               name,
+                               url,
+                               category,
+                               languages,
+                               paid,
+                               notes)
+    assert (response.status_code == 200)
+
+    # Resource not found
+    response = client.put("/api/v1/resources/0",
+                          json=dict(name="New name"),
+                          headers={'x-apikey': apikey},
+                          follow_redirects=True
+                          )
+    assert (response.status_code == 404)
+
+
+def test_commit_errors(
+        module_client, module_db, fake_auth_from_oc,
+        fake_commit_error, fake_algolia_save):
+    client = module_client
+    apikey = get_api_key(client)
+
+    response = client.put(
+        "/api/v1/resources/1",
+        json=dict(name="New name"),
+        headers={'x-apikey': apikey}
+    )
+    assert (response.status_code == 500)
+
+    response = create_resource(client, apikey)
+    assert (response.status_code == 500)
+
+
+def test_bad_requests(module_client, module_db, fake_auth_from_oc, fake_algolia_save):
+    client = module_client
+
+    apikey = get_api_key(client)
+    bad_json = "{\"test\": \"bad json\" \"test2\": \"still bad\"}"
+    response = client.post(
+        "api/v1/resources",
+        data=bad_json,
+        content_type='application/json',
+        headers={'x-apikey': apikey},
+        follow_redirects=True
+    )
+    assert (response.status_code == 400)
+    assert (isinstance(response.json, dict))
+
+
+def test_validate_resource(module_client, module_db, fake_auth_from_oc):
```

Comment:
There may be a better abstraction for this test rather than grouping the create and update routes together in one validate_resource test... but maybe not. Maybe we have a file called `test_validations`? I'm not sure, open to thoughts, but I don't think this belongs with anything else in this file in its current state. And as I stated above, I'd like to see the update and create tests broken out into two files, I'm not sure how this function will fit into that